### ISSUE TITLE
Terrible sound quality in Mac OS 10.15.6

### ISSUE THREAD

**Issue by kelheor** [daliansky/XiaoMi-Pro-Hackintosh#446]
After update I found that sound quality is terrible - like in cheap 1$ speakers. This issue happens only on Mac OS - no issues on Windows (I have dual boot). I tried to update EFI and install ALCPlugFix - it didn't help. Maybe somebody have an idea on how to fix that?

**Comment by stevezhengshiqi** [comment]
Currently, there’s no good solution for sound quality issues. You could try different layout-ids or using third party apps like Boom 3D.

**Comment by kelheor** [comment]
Do you also have such an issue? Or it randomly occurs only on certain laptops?

**Comment by MarFre22** [comment]
Hi,

I have been using Boom 3D, which have a similar sound quality to Dolby Atmos (I think), but only boost the sound quality on Headphones/earphones through Jack port.
After testing some audio "boost" apps, neither of them improve the speakers sound. Like you said, I can confirm the sound it's horrible (was always) really worse than on windows.

Hope for some help.

**Comment by stevezhengshiqi** [comment]
> Do you also have such an issue? Or it randomly occurs only on certain laptops?

There are situations that sound volume is smaller after the system update on my device. It’s not a problem for me since I am not heavily dependent on the speaker.

@MarFre22 I am pretty sure that no one would help on the sound issue. You could have a research on it on your own. I shared my development process on https://github.com/daliansky/XiaoMi-Pro-Hackintosh/issues/96

**Comment by Menchen** [comment]
@MarFre22 You should check your speaker (Build-In Output) in Audio MIDI Setup app (It's a system app that comes with macOS). It should be 2 channel 24-bit  48kHz.
![image](https://user-images.githubusercontent.com/6707265/89240676-21478980-d5fd-11ea-8208-41ac46c9ef4e.png)

On my testing the audio quality is good. I don't really remember how it was on Windows as I only used 2 days...
But the difference should be only post-processing...

**Comment by MarFre22** [comment]
@stevezhengshiqi 
Thank you for referring your develop, I forgot it in the middle of so many issues. Thanks to you (and Menchen of course) for your heavy work!

@Menchen 
I don't know how I didn't have that option like yours, thanks!  Tomorrow I will test better.
Yes, it's a post-processing thing.  In windows, without Dolby Atmos, the sound quality is more or less the same as the default in MacOS. But, for some reason, any sound enhancement app that, I tested, can't do post-processing on the speakers  (headphones only).

**Comment by Menchen** [comment]
@MarFre22 It's in `/Applications/Utilities/Audio MIDI Setup.app` or `/System/Applications/Utilities/Audio MIDI Setup.app
` Make sure that no headphone or BT audio device is connected.

Also using audio post-processing software is not recommended... It "may" cause malfunction to ALCPlugFix(Or maybe it just a hibernation issue on my side)... And In my opinion, the default sound is more natural... with Apps like Boom3D the audio sound "Fake" and loses details...
![image](https://user-images.githubusercontent.com/6707265/89242961-0aa43100-d603-11ea-97d5-896a7caa7fe6.png)

**Comment by MarFre22** [comment]
@Menchen 
After some testing, the sound quality remains the same. I don't know why the audio enchantment apps doesn't work on speakers.

I haven't been affected by hibernation issues due to these apps (I think so...). In my case, I need it because it's the only way I could find in order to try to reduce the difference of sound quality between Windows+Dolby and Mac.

**Comment by kelheor** [comment]
Well, reupdating EFI again with script helped to return sound quality to previous level. No more crackling and hissing sounds even on default alert sound. Maybe something wasn't installed properly after update. I also set higher frequency (44.1 -> 48kHz) in MIDI app. 

@MarFre22 Did you tested sound on different surfaces? For example, sound on wooden table and sound on cooling pad will be different.

**Comment by MarFre22** [comment]
@kelheor 
Yes I tested on cooling pad and on my hands. Unfortunately, the sound on speakers with or without any audio enchantment app is the same, which is far worse in bass than Windows+Dolby Atmos (without Dolby atmos is more or less the same).

Thanks for mentioned that. I also have the crackling problem (after some time using the same EFI). Sometimes I can't fix it cleaning the kext cache only. Did you apply the audio fix again?

**Comment by kelheor** [comment]
@MarFre22 No, but as I understand, this fix is mostly for headphones and I'm not using them.

**Comment by MarFre22** [comment]
Sometimes my hackintosh takes a while to start the apps in the menu bar. I saw on the console app there is a flood of warnings related with Sound. Maybe a malfunction with ALCPlugFix that @Menchen said?
- With SoundSource:
erro	18:40:55.348526+0100	SoundSource	AddInstanceForFactory: No factory registered for id <CFUUID 0x600000a139e0> F8BB1C28-BAE8-11D6-9C31-00039315CD46
erro	18:40:55.382566+0100	SoundSource	 HALC_ShellDriverPlugIn::Open: Can't get a pointer to the Open routine
erro	18:40:55.382921+0100	SoundSource	 HALC_ShellDriverPlugIn::Open: Can't get a pointer to the Open routine

- With CoreAudio:
erro	18:40:55.628508+0100	coreaudiod	 HALS_Object_GetPropertyData_DAI32: the object does not implement the property
erro	18:40:55.629129+0100	coreaudiod	 HALS_Object_GetPropertyData_DAI32: the object does not implement the property

- With SystemSoundServer:
erro	18:40:55.635978+0100	systemsoundserverd	 HALC_ShellDriverPlugIn::Open: Can't get a pointer to the Open routine

- With Boom 3D:
erro	18:40:58.510847+0100	Boom 3D	 HALC_ShellDriverPlugIn::Open: Can't get a pointer to the Open routine
erro	18:40:58.511321+0100	Boom 3D	 HALC_ShellDriverPlugIn::Open: Can't get a pointer to the Open routine
erro	18:40:58.511696+0100	Boom 3D	 HALC_ShellDriverPlugIn::Open: Can't get a pointer to the Open routine
erro	18:40:58.577234+0100	Boom 3D	 HALC_ShellObject::HasProperty: initial inner method call failed, Error: 560947818 (<private>)
erro	18:40:58.577682+0100	Boom 3D	 HALC_ShellObject::HasProperty: initial inner method call failed, Error: 560947818 (<private>)
erro	18:40:58.578485+0100	Boom 3D	 HALC_ShellObject::HasProperty: initial inner method call failed, Error: 560947818 (<private>)
erro	18:40:58.578751+0100	Boom 3D	 HALC_ShellObject::HasProperty: initial inner method call failed, Error: 560947818 (<private>)
erro	18:40:58.579000+0100	Boom 3D	 HALC_ShellObject::HasProperty: initial inner method call failed, Error: 560947818 (<private>)
erro	18:40:58.579248+0100	Boom 3D	 HALC_ShellObject::HasProperty: initial inner method call failed, Error: 560947818 (<private>)

**Comment by s-monti97** [comment]
Hi,
I'm also getting this sound problem. It sounds like a computer closed inside a box. Any solution? I'm on mi notebook pro gtx version.

**Comment by stevezhengshiqi** [comment]
@MarFre22 That's because you are using third-party sound applications on a hackintosh. They are not designed to support Realtek sound cards on hacks.

@s-monti97 I am afraid not. Don't expect solutions from EFI.

**Comment by MarFre22** [comment]
> @MarFre22 That's because you are using third-party sound applications on a hackintosh. They are not designed to support Realtek sound cards on hacks.
> 
> @s-monti97 I am afraid not. Don't expect solutions from EFI.

@stevezhengshiqi , oh it changes everything, thank you for the clarification. Would be nice to make a emulated layer that can make realtek hardware more friendly to these apps.

**Comment by ManuGithubSteam** [comment]
@kelheor  @s-monti97 

I would recommend a fresh install of 10.15.7. I did that and do not have sound problems. I even recorded the Sound in Windows and played it back in Mac to compare (because i found that thread before even installing 15.7). I would say they sound exactly the same. However audiophiles maybe can hear a difference from 10 % or so but they sound good. Been listening to ACDC for hours now....

Your data is not lost so i would just go for it.

**Comment by Ab2774** [comment]
Hi, I don't know if this is will work for you guys or not but I'm just sharing my experience:
In my old laptop with ALC230 and Kaby Lake-U processor, the audio quality was fantastic, just as it is on Windows, unfortunately, over time, the chassis broke, the hinge got loose, and the screen also broke, so I figured that it's not worth fixing, instead, I had another laptop with ALC236 and Kaby Lake-R processor, I transferred all my data to it and it's working fine, the only problem that I noticed is that the audio quality was so bad, the bass was unclear, I tried all of the layouts but all came with the same result, however, I copied the data from the Info.plist of ALC230 layout 20 to the Info.plist of ALC236 layout 14 as an experiment, unexpectedly, the quality changed and it sounded just like it was intended, so, I'm gonna create a new layout which has the improved Info.plist.
The bottom line is that invalid values in the Info.plist can result in a terrible quality sound and changing them can resolve them.
Cheers!

**Comment by Kyeanu** [comment]
if you're still looking for solutions, after getting a slow and bad audio and got boom audio not working anymore since Big Sur Update, I finally found a small soft called SpeakerAmps which is brilliant to get a more normal audio volume and quality, just play with settings until you understand and fine a good setup
